src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -420,14 +444,29 @@ internal void InvokeResponse(InvocationResponse invokeResponse)
                 try
                 {
                     IDictionary<string, object> bindingsDictionary = invokeResponse.OutputData
-                        .ToDictionary(binding => binding.Name, binding => binding.Data.ToObject());
+                        .ToDictionary(binding => binding.Name, binding => GetBindingData(binding));
 
                     var result = new ScriptInvocationResult()
                     {
                         Outputs = bindingsDictionary,
                         Return = invokeResponse?.ReturnValue?.ToObject()
                     };
                     context.ResultSource.SetResult(result);
+
```

Comment:
move freeing and sending `closeSharedMemoryResourcesRequest` to a separate method. Also this method needs to be called only if shared memory is used by an invocation. Check if `SharedMemorymanager` is tracking this invocation before calling free.

Reply:
Also, Rename `closeSharedMemoryResourcesRequest` to `FreeSharedMemoryResourcesRequest` ?

Reply:
Technically, it may not be freeing. It only closes the reference it has open. When we add some caching logic to this, we will potentially use this to make the worker close its references (but not free anything) since the host will have references to the same memory map (which we want to keep alive.)

I am going to change the message a little and call it `CloseSharedMemoryMaps` and pass it a list of maps we want to close.